[IV-23-24] Objetivo 2

shvtwp opened the pull request:
Sobre la estructura del repositorio

* [ ] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [ ] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

Sobre el análisis del problema

* [ ] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [ ] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [ ] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [ ] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

shvtwp committed:
Update .gitignore

Actualizado según el lenguaje de programación acordado

shvtwp committed:
Establecido lenguaje de programación closes marinajcs#6

shvtwp committed:
Añadida la entidad: Horario

shvtwp committed:
Añadida la entidad que modela el horario

Se ha implementado el fichero que modela el horario, el cual será también la representación del piso, incluyendo la lista de compañeros y las zonas comunes close #14

shvtwp committed:
Corregida la ubicación de la carpeta src

shvtwp committed:
Añadida la clase que modela las zonas comunes

El fichero añadido modela las zonas comunes, representando también los propietarios de cada una y una lista de tareas, donde se indica el nombre, la duración de cada una y si está o no realizada close #10

shvtwp committed:
Añadida la clase que define a un compañero de piso

En este fichero se modela lo que es un compañero de piso, con su identificador y horario personal (los intervalos horarios en los que no se encuentra disponible). Finalmente las preferencias quedan condicionadas por la pertenencia o no a determinada zona común, reflejado en src/ZonasComunes.ts close #8

shvtwp commented:
Sobre la estructura del repositorio

* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

Sobre el análisis del problema

* [ ] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [ ] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [ ] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

shvtwp commented:
@marinajcs Hola Marina, ¿podrías añadir el PR al M0 y añadir la etiqueta "hacktoberfest-accepted"? Está listo para que lo revises  👍

MarioGuisado reviewed (changes requested):
TypeScript necesita de un fichero package.json que deberás generar con objeto de mantener un historial de los paquetes instalados y optimizar la forma en que se generan las dependencias del proyecto.
En cuanto a las interfaces, siguiendo las buenas prácticas deberían describirse todas juntas en un único fichero, que será el que se importe cada vez que se necesite una. De esta manera se agrupan todas y no se deben ir buscando en los demás ficheros del proyecto cuando se necesiten.
Por otro lado intenta eliminar todo lo comentado, ya sea porque no se utiliza (será innecesario mantenerlo por tanto), o porque esté explicando alguna línea de código, algo que no debería ser necesario ya que el código en sí mismo se construye a base de pequeños commits autoexplicativos.

shvtwp committed:
Actualizados los nombres de los ficheros

shvtwp committed:
Delete src/Compi.ts

shvtwp committed:
Delete src/Horario.ts

shvtwp committed:
Delete src/ZonasComunes.ts

shvtwp committed:
Añadida una clase que agrupa las interfaces

shvtwp committed:
Añadido package.json

shvtwp committed:
Comentarios eliminados

MarioGuisado reviewed (approved):
Como apunte que no se olvide asignar todos los issues a un milestone (en este caso el 0), creo que queda por asignar el correspondiente a la elección de lenguaje.

shvtwp committed:
Renombra la carpeta src a lib

shvtwp committed:
Elimina el fichero dedicado a las interfaces

Las interfaces pasan a ser tipos expresados en el fichero de la clase donde se emplean

shvtwp committed:
Actualiza el path de la entidad

marinajcs commented:
* [x] ¿Los issues que ha planteado la persona son efectivamente problemas y
evidencian comprensión de las historias de usuario? Si no es así, es mejor que
se le transmita antes de que empiece a crear cualquier código erróneo.
* [x] ¿Has comprobado que efectivamente el compañero/a ha seguido *su* lista de
      comprobación?
* [x] ¿El código creado es una modelización válida de las historias de usuario
      creadas por ti, y te sirve como base para programar la lógica de negocio?
* [x] ¿El código creado sigue las mejores prácticas para el lenguaje de
      programación que hayáis consensuado, tanto en identificadores, como
      nombres de ficheros, como disposición de los diferentes ficheros en el
      repositorio? Por ejemplo, ¿todos los commits se refieren a un issue y
      aportan código para resolver el problema planteado en el mismo?
* [x] ¿He comprobado que el código compile, o que quien lo ha programado ha lo
ha hecho y documentado en el repositorio?
* [x] ¿He añadido el PR y todos los issues que se resuelvan en él al milestone
correspondiente, que será el primero?
* [x] ¿Cubre totalmente el milestone 0, o hará falta modelización adicional en
el objetivo 4 para poder programar la lógica de negocio?
* [x] En resumen, ¿considero que con lo hecho puedo ponerme a trabajar
directamente en el siguiente milestone, que incluye la lógica de negocio?

shvtwp committed:
Actualiza los constructores

Los constructores ahora son la versión corta

shvtwp commented:
@JJ listo para revisión.

JJ commented:
Se me había pasado totalmente, perdona.

JJ commented:
Tendría que aprobarlo explícitamente @marinajcs antes, por eso no lo he revisado. Por favor, hacedlo lo antes posible.

marinajcs reviewed (approved):
Me parece una buena modelización del problema

JJ commented:
Voy a revisarlo, pero en el futuro, por favor, pedídmelo explícitamente.

JJ reviewed (changes requested):
Me temo que no hay ni un commit que siga las mejores prácticas. No tienen la longitud correcta (50 caracteres en la primera) y mencionan el issue con el que trabajan al final, con lo que es complicado saber si se están siguiendo las buenas prácticas o no. Dejadme que le dé otro repaso para poder  ver si realmente corresponde a las HUs.

JJ commented on lib/compi.ts:
```diff
@@ -0,0 +1,12 @@
+
+export class Compi {
+    constructor(
+        readonly id: number,
```
Los objetos valor nunca tienen un ID. Pueden usar un string para identificarlos, pero no necesariamente dentro de un objeto. Como he dicho en #8, empezar por una estructura determinada te fuerza a crearla; si te fijas en qué problema hay o cómo se van a usar en la lógica de negocio saldrán de forma natural las estructuras necesarias.
En todo caso, el issue #8 tampoco dice nada de un ID.

JJ commented on lib/horario.ts:
```diff
@@ -0,0 +1,9 @@
+
+class Horario {
+    constructor(
+        private readonly idpiso: number, 
```
Bien por el uso de la forma condensada de los constructores de TS, pero una vez más, ¿qué uso va a darle la lógica de negocio al id?

JJ commented on lib/zonascomunes.ts:
```diff
@@ -0,0 +1,16 @@
+    realizada: boolean;
+}
+
+export class ZonasComunes {
```
A través de los cambios de nombre de fichero he perdido cuál es el issue que se está desarrollando aquí. Es importante que siempre, incluso en los cambios, se ponga el issue porque es documentación para la persona que revisa y saber si está bien o mal.

JJ commented on lib/zonascomunes.ts:
```diff
@@ -0,0 +1,16 @@
+type Tareas = {
+    tarea: string;
+    duracionEstimada: number;
+    realizada: boolean;
```
Los objetos valor deben ser inmutables.  Hay otra forma de gestionar si una tarea se ha realizado o no, usando entidades. Convendría que consideraras cómo hacerlo de esa forma.

JJ reviewed (changes requested):
OK, ya he visto en algunos casos cómo se ha desarrollado, pero en otros es imposible saberlo. En todo caso, tanto @marinajcs lo que tienen que hacer es una especie de recorrido como 
1. Tenemos unos compañeros de piso con unos horarios 
2. Hay unas tareas que tienen ¿horario?¿prioridad? 
3. Las tareas una vez realizadas ¿se almacenan en algún sitio? ¿se borran?

Convendría quizás replantearse los "issues-son-objetos-o-entidades" y convertirlos en "issues-son-problemas". Como no está claro qué código se ha desarrollado con respecto a qué issue, en algunos casos se puede reescribir; en otros habrá que marcarlo como inválido y cerrarlo cuando se fusione el PR.

shvtwp committed:
Elimina la clase zonascomunes.ts close marinajcs#10

No será necesaria, no aporta nada a la lógica de negocio

shvtwp committed:
Modifica la representación de un compañero #8

Se elimina la id porque no aporta nada a la lógica de negocio, además, dejan de tener asignado un horario de disponibilidad que pasa a ser un número determinado de horas semanales disponibles close marinajcs#8

shvtwp committed:
Modela las tareas domésticas #16

Se tipifica qué es una tarea

shvtwp committed:
Cambio de nombre a tareas.ts #16

Para que no haya confusión entre el tipo tareas y la clase, se ha renombrado la clase como reparto

shvtwp committed:
Modela la gestión de las tareas #16

Se almacenan las tareas en dos vectores, uno de tareas completadas y otro de tareas disponibles. La idea es usarlos en la asignación de las tareas. Se crea también un método para establecer el número de puntos que cada compañero tiene que alcanzar semanalmente, teniendo en cuenta que se ha orientado el problema de manera en que el reparto se hace en función del tiempo y la dificultad, a modo de conseguir un reparto equitativo closes marinajcs#16

shvtwp commented:
@marinajcs He hecho nuevas modificaciones significativas, así que creo que deberías volver a revisar los cambios antes de solicitar revisión

marinajcs commented on lib/reparto.ts:
```diff
@@ -0,0 +1,38 @@
+import { Compi } from './compi';
+
+type Tarea = {
```
Por lo que tengo entendido, puntuación es el nivel de dificultad de la tarea, lo que me parece buena idea de cara a realizar una asignación más equitativa.

Como sugerencia, se podría añadir a las tareas una propiedad opcional: `personasNoPermitidas?: Compi[];`  
Así, también podría abordarse la HU02 #4

marinajcs replied:
Adicionalmente, el nombre de la tarea no es necesario para implementar la lógica de negocio

marinajcs reviewed (changes requested):
Exceptuando dos pequeños detalles, me parece un modelo del problema correcto y una buena base para las futuras etapas de desarrollo.

marinajcs commented on iv.yaml:
```diff
@@ -0,0 +1,2 @@
+lenguaje: TypeScript
+entidad: lib/Horario.ts
```
También faltaría actualizar la entidad a `lib/reparto.ts`

shvtwp committed:
Actualiza entidad

shvtwp committed:
Elimina el nombre de la tarea #16

No contribuye a la lógica de negocio

shvtwp committed:
Añade atributos para gestionar preferencias #17

Si un compañero prefiere una tarea, en la asignación tratará de buscar primero entre esas opciones, después entre las neutras y finalmente entre las detestadas. Ya no está planeado el issue close marinajcs#14

shvtwp commented:
@marinajcs He actualizado según los cambios que me has dicho. En lugar de añadir una representación de las preferencias en Tarea, la he añadido en los propios Compis con el fin de representar de manera más intuitiva las preferencias personales. Revísalo y comunícame si tienes alguna otra sugerencia.

marinajcs reviewed (approved):
Me parece bien 👍 
Puedes referenciar a los commits que resuelven los comentarios y cerrarlos/resolverlos para despejar el PR

marinajcs commented:
Listo para revisión @JJ

JJ reviewed (approved):
:+1: 
Quizás no deberías haber incluido ningún setter, pero si @marinajcs lo ha aprobado, me vale.
